### PR TITLE
Issue#66 important sync with be

### PR DESCRIPTION
**Pull request by tania833**
develop

## GitHub Board

**Issue link**

[#66  ]( important sync with be )

### CODE REVIEW

File: src/modules/main/store/mainSlice.ts [hunk 1]
```diff
@@ -58,8 +59,9 @@ export default mainSlice.reducer;
 
 export const fetchImportantPosts = (): AppThunkType => async (dispatch) => {
   try {
-    const posts = await Promise.resolve(MOCK_POSTS);
-    dispatch(loadImportant(posts));
+    const posts = await Promise.resolve(getPosts('important'));
```
Comment (calster00) [inline src/modules/main/store/mainSlice.ts:62]: `getPosts` already returns a promise
```suggestion
    const posts = await getPosts('important');
```

Comment (tania833) [reply src/modules/main/store/mainSlice.ts:62]: Thank you 💪🏻

File: src/modules/main/store/mainSlice.ts [hunk 2]
```diff
@@ -77,7 +79,7 @@ export const fetchExperts = (): AppThunkType => async (dispatch) => {
 export const fetchNewestPosts = (): AppThunkType => (dispatch, getState) => {
   const { meta } = getState().main.newest;
 
-  const totalNewestPosts =  meta.totalNewestPosts || MOCK_NEWEST.length;
+  const totalNewestPosts = meta.totalNewestPosts || MOCK_NEWEST.length;
```
Comment (SergeOlabin) [inline src/modules/main/store/mainSlice.ts:82]: no mock needed

File: src/modules/main/store/mainSlice.ts [hunk 3]
```diff
@@ -21,7 +22,7 @@ interface INewestPostPayload {
 
 export interface IMainState {
   newest: INewestPostPayload;
-  important: IPost[];
+  important: PostResponseType[];
```
Comment (SergeOlabin) [inline src/modules/main/store/mainSlice.ts:25]: should be still IPost[],
you can align IPost to have the same properties as on the response

**Review by SergeOlabin — CHANGES_REQUESTED** [review]

File: src/modules/main/store/mainSlice.ts (hunk 2 above)
Comment (tania833) [reply src/modules/main/store/mainSlice.ts:82]: My section is important. Do I have to delete it for newest?

File: src/modules/main/store/mainSlice.ts (hunk 3 above)
Comment (tania833) [reply src/modules/main/store/mainSlice.ts]: done

File: src/modules/main/store/mainSlice.ts [hunk 4]
```diff
@@ -58,8 +58,30 @@ export default mainSlice.reducer;
 
 export const fetchImportantPosts = (): AppThunkType => async (dispatch) => {
   try {
-    const posts = await Promise.resolve(MOCK_POSTS);
-    dispatch(loadImportant(posts));
+    const posts = await getPosts('important');
+    const loadedPosts = posts.data.content.map((post) => {
+      return {
+        author: {
+          photo: post.author.avatar,
```
Comment (SergeOlabin) [inline src/modules/main/store/mainSlice.ts:65]: let's have the same property names  - `avatar` and `lastName`.
you can use https://lodash.com/docs/4.17.15#pick here.

e.g.
```
author: {
    ...pick(post.author, [
        avatar,
        firstName,
        lastName,
        id,
        mainInstitution
    ]),
}
```

**Review by SergeOlabin — CHANGES_REQUESTED** [review]

File: src/modules/main/store/mainSlice.ts (hunk 4 above)
Comment (tania833) [reply src/modules/main/store/mainSlice.ts]: done

**Review by SergeOlabin — APPROVED** [review]

**Pull request merged**